### PR TITLE
IEP-187: Add project name to size analysis editor (IEP-291)

### PR DESCRIPTION
**Pull request by sigmaaa**
Now ESP-IDF Application size analysis editor is showing the project name
IEP-187

### CODE REVIEW

**Comment by kolipakakondal** [comment]
Hi @sigmaaa 

Adding the name in the Overview section leads to space issues, as we have limited space. Can you replace the editor tab name with the project name? Which is similar to any Java editor.

**Comment by sigmaaa** [comment]
> Hi @sigmaaa
> 
> Adding the name in the Overview section leads to space issues, as we have limited space. Can you replace the editor tab name with the project name? Which is similar to any Java editor.

Please check the new commit. Now it's looks like this: 
IEP-187new

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeAnalysisEditor.java [hunk 1]
```diff
@@ -37,7 +37,8 @@ protected void createPages()
 		String osString = file.getLocation().toOSString();
 		project = file.getProject();
 		Logger.log("Editor input:" + osString); //$NON-NLS-1$
-
+		
+		setEditorTabTitle(project.getName());
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeAnalysisEditor.java:41]: This is good, but the ideal way to set the setPartName(..) is part of the MultiPageEditorPart.init(...) 

You can override the init(...) implementation and then set the part name setPartName(..) directly without calling a different method again

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeAnalysisEditor.java [hunk 2]
```diff
@@ -122,4 +123,7 @@ private String getTarget()
 		return new SDKConfigJsonReader(project).getValue("IDF_TARGET");
 	}
 
+	private void setEditorTabTitle(String title) {
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeAnalysisEditor.java:126]: You can avoid this method

**Comment by kolipakakondal** [comment]
LGTM. Approved

**Review by kolipakakondal — APPROVED** [review]

**Pull request merged**